sky/server/requests/queues/mp_queue.py

```diff
@@ -32,7 +32,16 @@ def queue_getter(q_obj):
         QueueManager.register(name, callable=queue_getter(q_obj))
 
     # Start long-running manager server.
-    manager = QueueManager(address=('', port), authkey=b'skypilot')
+    # Manager will set socket.SO_REUSEADDR, but BSD and Linux have different
+    # behaviors on this option:
+    # - BSD(e.g. MacOS): * (0.0.0.0) does not conflict with other addresses on
+    #   the same port
+    # - Linux: in the contrary, * conflicts with any other addresses
+    # So on BSD systems, binding to * while the port is already bound to
+    # localhost (127.0.0.1) will succeed, but the Manager won't actually be able
+    # to accept connections on localhost.
+    # For portability, we use the loopback address instead of *.
+    manager = QueueManager(address=('localhost', port), authkey=b'skypilot')
```

Comment:
Still have a problem, if the port is already bound by other applications, `get_queue` might hang forever when connect to the used port, which cause a server hang.

Reply:
fixing

Reply:
Would it be possible to explicitly check if the port is occupied and error out before we start the manager?

Reply:
Maybe we don't have to bother checking potential causes of broken QueueManager, adding a timeout when connects to the QueueManager will cover the overall availability.

Reply:
Oh, yes, timeout is not enough. The server using the port might respond, which is even worse.